src/listener.rs

```diff
@@ -22,9 +22,12 @@ fn handle_new_conn(new_conn: quinn::NewConnection) {
     // send in their certificate - have a delay and then log and forget them
     let q_conn = new_conn.connection;
     let incoming_streams = new_conn.incoming;
-
     let peer_addr = q_conn.remote_address();
 
+    current_thread::spawn(new_conn.driver.map_err(move |e| {
+        communicate::handle_communication_err(peer_addr, &From::from(e), "Connection driver failed")
```

Comment:
This should do what this is doing:
https://github.com/ustulation/using-quinn/blob/df0d633103f579c5b1bcb84dd600d345c35a52f8/src/listener.rs#L88
not call anything from the `communicate` module as that also fires an event which we might not need (if someone was incoming and got disconnected before doing anything, user didn't even know about it in the first place so no use firing a `ConnectionFailure` event to them)

You can create an extra function here to direct both of these (the one above and the one linked) to that if you want to print it but do keep it confined to the module (in this case `listener.rs`). Then later it makes it easier to analyse if we can combine something.